torch/csrc/autograd/engine.cpp

```diff
@@ -422,6 +510,10 @@ auto Engine::evaluate_function(FunctionTask& task) -> void {
       }
     }
   }
+
+  #ifdef WITH_CUDA
+    if (has_event) THCudaCheck(cudaEventDestroy(event));
```

Comment:
We should have a RAII wrapper around `cudaEvent_t`

Reply:
See above but sounds like a good idea and it should dramatically reduce the apparent complexity.